* Do not use a fixed lookup table, to allow other unrecognised cards
  to possibly work
diff -r c34020f29f4a libdha/pci_db2c.awk
--- a/vidix/pci_db2c.awk	Sun Apr 30 16:54:29 2006 +0200
+++ b/vidix/pci_db2c.awk	Sun Apr 30 16:54:30 2006 +0200
@@ -60,9 +60,6 @@ BEGIN {
     print_name_struct(name_h_file);
     print "#include <stddef.h>">name_file
     print "#include \"pci_names.h\"">name_file
-    print "#include \"pci_dev_ids.c\"">name_file
-    print "">name_file
-    print "static struct vendor_id_s vendor_ids[] = {">name_file
     first_pass=1;
     init_name_db();
     while(getline <in_file) 
@@ -78,7 +75,6 @@ BEGIN {
 		printf("#define VENDOR_%s\t", svend_name) >vendor_file;
 		if(length(svend_name) < 9) printf("\t") >vendor_file;
 		printf("0x%s /*%s*/\n",field[2], name_field) >vendor_file;
-		printf("{ 0x%s, \"%s\", dev_lst_%s },\n",field[2], name_field, field[2]) >name_file;
 		printf("/* Vendor: %s: %s */\n", field[2], name_field) > ids_file
 		if(first_pass == 1) { first_pass=0; }
 		else	{ print "{ 0xFFFF,  NULL }\n};" >dev_ids_file; }
@@ -118,7 +113,6 @@ BEGIN {
     print "#endif/*PCI_IDS_INCLUDED*/">ids_file
     print "">name_h_file
     print "#endif/*PCI_NAMES_INCLUDED*/">name_h_file
-    print "};">name_file
     print "{ 0xFFFF,  NULL }" >dev_ids_file;
     print "};">dev_ids_file
     print_func_bodies(name_file);
@@ -172,30 +167,11 @@ function print_func_bodies(out_file)
    print "">out_file
    print "const char *pci_vendor_name(unsigned short id)" >out_file
    print "{" >out_file
-   print "  unsigned i;" >out_file
-   print "  for(i=0;i<sizeof(vendor_ids)/sizeof(struct vendor_id_s);i++)">out_file
-   print "  {" >out_file
-   print "\tif(vendor_ids[i].id == id) return vendor_ids[i].name;" >out_file
-   print "  }" >out_file
    print "  return NULL;" >out_file
    print "}">out_file
    print "" >out_file
    print "const char *pci_device_name(unsigned short vendor_id, unsigned short device_id)" >out_file
    print "{" >out_file
-   print "  unsigned i, j;" >out_file
-   print "  for(i=0;i<sizeof(vendor_ids)/sizeof(struct vendor_id_s);i++)">out_file
-   print "  {" >out_file
-   print "\tif(vendor_ids[i].id == vendor_id)" >out_file
-   print "\t{" >out_file
-   print "\t  j=0;" >out_file
-   print "\t  while(vendor_ids[i].dev_list[j].id != 0xFFFF)" >out_file
-   print "\t  {">out_file
-   print "\t\tif(vendor_ids[i].dev_list[j].id == device_id) return vendor_ids[i].dev_list[j].name;">out_file
-   print "\t\tj++;">out_file
-   print "\t  };">out_file
-   print "\t  break;" >out_file
-   print "\t}" >out_file
-   print "  }" >out_file
    print "  return NULL;">out_file
    print "}">out_file
    return
